### PR TITLE
Java demo

### PR DESCRIPTION
**Pull request by svangasse**
I've implemented a duplicate of the existing webrtc demos using the Java GStreamer bindings. I've added build and run instructions to the readme, and also added it as an option in the docker-compose file

### CODE REVIEW

File: sendrecv/gst-java/src/main/java/WebrtcSendRecv.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    private WebRTCBin webRTCBin;
+    private Pipeline pipe;
+
+    public static void main(String[] args) throws Exception {
+        if (args.length == 0) {
+            logger.error("Please pass a session id from the signalling server: java -jar build/libs/gst-java.jar 1234");
+            return;
+        }
+        String sessionId = args[0];
+        String serverUrl = args.length == 2 ? args[1] : REMOTE_SERVER_URL;
```
Comment (MathieuDuponchelle) [inline sendrecv/gst-java/src/main/java/WebrtcSendRecv.java]: Can you ideally accept the server url in the same form as the other examples? (`--server SERVER`)

Comment (svangasse) [reply sendrecv/gst-java/src/main/java/WebrtcSendRecv.java]: No problem. Done

Comment (MathieuDuponchelle) [reply sendrecv/gst-java/src/main/java/WebrtcSendRecv.java]: awesome :)

**Comment by MathieuDuponchelle** [comment]
Giving it a spin, this looks promising!

File: docker-compose.yml [hunk 2]
```diff
@@ -5,8 +5,10 @@ services:
   #
   # sendrecv-gst:
   #   build: ./sendrecv/gst
-  sendrecv-gst-rust:
-    build: ./sendrecv/gst-rust
+  sendrecv-gst-java:
+    build: ./sendrecv/gst-java
+  #sendrecv-gst-rust:
```
Comment (MathieuDuponchelle) [inline docker-compose.yml:10]: why do you comment this out?

File: README.md [hunk 3]
```diff
@@ -92,6 +92,12 @@ With all versions, you will see a bouncing ball + hear red noise in the browser,
 
 You can pass a --server argument to all versions, for example `--server=wss://127.0.0.1:8443`.
 
+#### Running the Java version
+
+`cd sendrecv/gst-java`\
+`./gradlew build`\
+`java -jar build/libs/gst-java.jar [session id]` with id from browser
```
Comment (MathieuDuponchelle) [inline README.md]: You need to update this :)

**Review by MathieuDuponchelle — COMMENTED** [review]
I can't run the example, I'm getting https://gist.github.com/MathieuDuponchelle/be0452d4fb0960a35222c821e31f1568 , certainly because of the self-signed certificate :) It's probably worth catching this exception, and updating the README to explain how to use a self-signed certificate or work around this

File: docker-compose.yml (hunk 2 above)
Comment (svangasse) [reply docker-compose.yml:10]: The docker-compose.yml file contains the comment: "uncomment the sendrecv you would like to use". I added the java lines uncommented for easier testing of this PR

File: README.md (hunk 3 above)
Comment (svangasse) [reply README.md]: Fixed

**Comment by svangasse** [comment]
I can't reproduce the self-signed certificate error you're seeing either on my local machine or on the docker container included in the example. It might be something specific to your Java setup, around your local keystore or truststore. I've added some code that I think should force it to use an insecure trust manager. Can you give that a try?

**Comment by MathieuDuponchelle** [comment]
This works great now :D

I did have to change the gradle `distributionUrl` to `https\://services.gradle.org/distributions/gradle-5.2.1-bin.zip`, I couldn't build with 4.6 and java 11. Can you upgrade to this gradle version, or did you have a specific reason for using 4.6?

I have no other remarks, note that I didn't review the java code in depth, not really familiar with this language :)

**Comment by saket424** [comment]
@svangasse 
Have you tried compiling the java demo for android ? I'd be very interested in getting webrtcbin to run on android. This url shows a disussion thread of someone trying to sendrecv on workstation working with android chrome

http://gstreamer-devel.966125.n4.nabble.com/WebRTC-dtls-errors-when-sender-and-receiver-are-on-different-networks-tt4688758.html

Any pointers you can provide to get this to run on android are much appreciated

Thanks

**Comment by MathieuDuponchelle** [comment]
@saket424 for what it's worth I've already tested using webrtcbin through the android JNI and it worked just fine, if you don't mind using GStreamer's C API

**Comment by saket424** [comment]
@MathieuDuponchell,
I absolutely don't mind using the C api if it is already is known to work in Android. I'll open a new issue on the issue list so we move our discussion there.
Thanks

**Comment by ystreet** [comment]
puts https://github.com/ystreet/gstwebrtc-demos/tree/android here

**Comment by svangasse** [comment]
Updated gradle version as suggested

**Comment by MathieuDuponchelle** [comment]
Thanks @svangasse , nice work ! :tada:

**Comment by pradeep538** [comment]
Please help by giving the detail example of how to run the docker with gst-java
Thanks in advance.

**Pull request merged**